mars/dataframe/tests/test_core.py

```diff
@@ -254,3 +254,88 @@ def test_key_value(setup):
 
     result = idx.values.execute().fetch()
     np.testing.assert_array_equal(result, raw.values)
+
+
+def test_between(setup):
+    pd_series = pd.Series(pd.date_range("1/1/2000", periods=10))
+    pd_left, pd_right = pd_series[2], pd_series[7]
+    expected = pd_series.between(pd_left, pd_right)
+
+    series = Series(pd_series, chunk_size=5)
+    left, right = series[2], series[7]
+    result = series.between(left, right).execute().fetch()
+    pd.testing.assert_series_equal(result, expected)
+
+
+def test_between_datetime_values(setup):
+    pd_series = pd.Series(pd.bdate_range("1/1/2000", periods=20).astype(object))
+    pd_series[::2] = np.nan
+    expected = pd_series[3:18].dropna()
+
+    series = Series(pd_series, chunk_size=5)
+    result = series[series.between(series[3], series[17])]
+    result = result.execute().fetch()
+    pd.testing.assert_series_equal(result, expected)
+
+    result = series[series.between(series[3], series[17], inclusive="neither")]
+    result = result.execute().fetch()
+    expected = pd_series[5:16].dropna()
+    pd.testing.assert_series_equal(result, expected)
+
+
+def test_between_period_values(setup):
+    pd_series = pd.Series(pd.period_range("2000-01-01", periods=10, freq="D"))
+    pd_left, pd_right = pd_series[2], pd_series[7]
+    expected = (pd_series >= pd_left) & (pd_series <= pd_right)
```

Comment:
Use `between` in pandas to calculate expected result.

Reply:
Done